frontend/packages/client/src/components/Community/JoinCommunityButton.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from "react";
+import { useWebContext } from "../../contexts/Web3";
+import { useJoinCommunity, useUserRoleOnCommunity } from "../../hooks";
+
+export default function JoinCommunityButton({
+  enableJoin = false,
+  alignment = "flex-end",
+  communityId,
+  setTotalMembers = () => {},
+}) {
+  const [joinedBtnText, toggleJoinedBtnText] = useState("Joined");
+  const [borderColor, setBorderColor] = useState("transparent");
+  const [isMember, setMemberState] = useState(false);
+  const { createCommunityUser, deleteUserFromCommunity } = useJoinCommunity();
+  const { injectedProvider, user } = useWebContext();
+  const { addr } = user;
+  const memberState = useUserRoleOnCommunity({
+    addr,
+    communityId,
+    roles: ["member"],
+  });
+
+  useEffect(() => {
+    setMemberState(memberState);
+  }, [memberState]);
+
+  const handleHover = () => (isMember ? hoverToggle() : () => {});
+
+  const hoverToggle = () => {
+    if (joinedBtnText === "Joined") {
+      toggleJoinedBtnText("Leave");
+      setBorderColor("#FBD84D"); // defined in App.sass
+    } else {
+      toggleJoinedBtnText("Joined");
+      setBorderColor("transparent");
+    }
+  };
```

Comment:
Could we simplify the button so that we only have Join or Leave? This would simplify the logic needed and make the action of the button very clear without needing to hover to understand the action that will be performed if clicked.

Reply:
Confirmed we can do this, and implementing the change.